rte/kernels/mo_rte_solver_kernels.F90

```diff
@@ -871,8 +871,10 @@ pure subroutine sw_dif_and_source(ncol, nlay, top_at_1, mu0, sfc_albedo, &
         source_dn(i,lay_index) =    Tdir * dir_flux_inc(i)
         dir_flux_trans(i)      = Tnoscat * dir_flux_inc(i)
       end do
+      if (j==nlay) then
```

Comment:
Hi @MHBalsmeier Thanks for these suggestions. This particular suggestion I don't want to adopt because it puts an if-test inside a loop, which may degrade performance quite a bit. We're using a `for` loop precisely because there's a loop carry dependency, so the transmitted direct-beam flux is guaranteed to be correct at the end of the loop.

Reply:
Yes I assumed this already but the compiler gave a warning when compiled with `-Wall`. I undid it.